runelite-client/src/main/java/net/runelite/client/plugins/discord/DiscordState.java

```diff
@@ -203,8 +201,25 @@ void triggerEvent(final DiscordGameEventType eventType)
 	 */
 	void checkForTimeout()
 	{
+		if (events.isEmpty())
+		{
+			return;
+		}
+
 		final Duration actionTimeout = Duration.ofMinutes(config.actionTimeout());
 		final Instant now = Instant.now();
+		final EventWithTime eventWithTime = events.get(0);
+
 		events.removeIf(event -> event.getType().isShouldTimeout() && now.isAfter(event.getUpdated().plus(actionTimeout)));
+
+		if (eventWithTime.getType().getState().equals(DiscordGameEventType.IN_MENU.getState()) && now.isAfter(eventWithTime.getStart().plus(actionTimeout)))
+		{
+			final DiscordPresence presence = lastPresence
+				.toBuilder()
+				.startTimestamp(null)
+				.build();
+			lastPresence = presence;
+			discordService.updatePresence(presence);
```

Comment:
On second thought, shouldn't this be calling `discordService.reset()` rather than clearing the last known time? We would like to remove RuneLite from showing as the active game if a user has been in menus for longer than their configured timeout, not just remove the timer showing how long they've been in menus.

Reply:
> This should probably just not display time anymore instead of removing entire rich presence

If I'm not mistaken deathbeam had said not to do that. This was what the original code had done but I changed that at his request.

Reply:
@deathbeam Can you comment on this? Your earlier comment on this PR is as DSerp says, but the issue this PR is addressing (which you created) contradicts that.

Reply:
My PR for this issue does what I commented, I just forgot to update the issue. I still like being in menu displayed, but being in menu for 9 hours is useless information